### PR TITLE
ISSUE-1240 WebSocket: ability to disable IP check for ticket authentication

### PR DESCRIPTION
**Pull request by quantranhong1999**
resolve https://github.com/linagora/tmail-backend/issues/1240

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
CI failed
`AIBotIntegrationTest.shouldReplyToRecipientsInToHeader(AIBotIntegrationTest.java:202)`
it looks not related to this pr

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/ticket/TicketManager.scala [hunk 1]
```diff
@@ -29,10 +32,23 @@ case class Ticket(clientAddress: InetAddress,
...
+  private val ipValidationEnabled: Boolean = true
+
+  def extractIpValidationEnableConfig(propertiesProvider: PropertiesProvider): Boolean =
+    try {
+      Option(propertiesProvider.getConfiguration("jmap"))
+        .map(config => config.getBoolean(ipValidationEnableProperty, ipValidationEnabled))
+        .getOrElse(ipValidationEnabled)
+    } catch {
+      case _: FileNotFoundException => ipValidationEnabled
+    }
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/ticket/TicketManager.scala:46]: Why not put that into JMAPExtensionConfiguration ?

Over places where we manually acces the JMAP conf file:

```
XXX:~/Documents/openpaas-james/tmail-backend/jmap/extensions$ grep 'getConfiguration("jmap")' * -r
src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventReplyPerformer.scala:    Try(propertiesProvider.getConfiguration("jmap"))
src/main/scala/com/linagora/tmail/james/jmap/method/EmailRecoveryActionSetMethod.scala:    val config = Try(propertiesProvider.getConfiguration("jmap"))
src/main/scala/com/linagora/tmail/james/jmap/oidc/WebFingerRoutes.scala:    Try(propertiesProvider.getConfiguration("jmap"))
```
Time for a refactoring?

Comment (quantranhong1999) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/ticket/TicketManager.scala:46]: I just see that. Let's refactor a bit there.

File: tmail-backend/jmap/extensions/pom.xml [hunk 2]
```diff
@@ -57,6 +57,12 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>james-server-guice-configuration</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-guice-common</artifactId>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
```
Comment (HoussemNasri) [inline tmail-backend/jmap/extensions/pom.xml:65]: This is not sorted correctly. It should come before `james-server-guice-configuration` as 'm' comes before 'n'.

**Comment by quantranhong1999** [comment]
Not related but likely a flaky test piped up https://james-jenkins.lin-saas.com/blue/organizations/jenkins/Tmail%20build/detail/PR-1263/5/tests

sendEmailShouldResultingInEventuallySavingDataToBothObjectStoragesWhenSecondStorageIsDownForShortTime(GuiceJamesServer) – com.linagora.tmail.james.DistributedLinagoraSecondaryBlobStoreTest
46s
Error
software.amazon.awssdk.core.exception.SdkClientException: Unable to execute HTTP request: Read timed out

cc @hungphan227

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
@quantranhong1999 squash maybe?

**Review by Arsnael — APPROVED** [review]

**Pull request merged**